src/firefly/js/ui/Menu.jsx

```diff
@@ -128,10 +134,15 @@ function AdjustableMenu({menuTabItems, helpItem, selected, dropDown, showUserInf
         if (ts && tabCount===-1) setTabCount(menuTabItems.length); // force a rerender when we know the html element of the tab bar
     },[]);
 
-    const setElement= useCallback( (key,e) => {
-        if (!e) return;
+    const setElement= useCallback( (key,el) => {
+        if (!el) return;
+
+        // save the first and last tab elements in the layout info store, needed for positioning the app hints
+        if (key==='0') dispatchUpdateLayoutInfo({menuTabNode: {first: el}});
+        if (key===menuTabItems.length-1+'') dispatchUpdateLayoutInfo({menuTabNode: {last: el}});
+
```

Comment:
I am concerned you are to aggressively calling calling redux here. We make too many layout info update anyway. I am sort of surprise it works at all but since it does here a a couple of sugestion.

- only call if the element not set
```js
  if (getLayouInfo().menuTabNode.last!==el) dispatchUpdateLayoutInfo({menuTabNode: {last: el}});
```
- set the first and last in one call.
   - set the first in a ref
   - make the call when you are ready to call the last then set both

Both of these change will drastically reduce the call to redux.

Reply:
I added a logpoint. The callback is being called 2 times for every button. So I think the dispatch is happening 4 times. All this happens for every tab click. Before there was not that much overhead, but if it is doing a dispatch then I am more concerned.

Reply:
Oh, I see - will make this change then